src/api/checkin.js

```diff
@@ -0,0 +1,19 @@
+import express from 'express';
+import { addAction } from '../lib/database';
+import CheckinStore from '../store/checkin-store.js';
+
+let app = express();
+
+app.post('/', (req, res) => {
+    addAction('CHECKIN', {
+        studentId: req.body.studentId,
+        itemAddress: req.body.itemAddress
+    }).then(actionId => {
+        CheckinStore.getCheckinByActionId(actionId);
```

Comment:
This is a check to see if the checkin succeeded. If it failed. getCheckinByActionId will fail, and the catch will kick in.